pyparsing/core.py

```diff
@@ -558,80 +566,32 @@ def postParse(self, instring, loc, tokenlist):
 
     # ~ @profile
     def _parseNoCache(self, instring, loc, doActions=True, callPreParse=True):
-        TRY, MATCH, FAIL = 0, 1, 2
-        debugging = self.debug  # and doActions)
-
-        if debugging or self.failAction:
```

Comment:
Instead of having two near-identical sections, we will add this check *if* an exception happens